src/runtime_src/core/tools/xbmgmt2/flash/firmware_image.cpp

```diff
@@ -459,6 +459,145 @@ std::ostream& operator<<(std::ostream& stream, const DSAInfo& dsa)
     return stream;
 }
 
+/**
+ * Helper method to find a set of bytes in a given buffer.
+ * 
+ * @param pBuffer The buffer to be examined.
+ * @param _bufferSize
+ *                The size of the buffer.
+ * @param _searchString
+ *                The string to search for.
+ * @param _foundOffset
+ *                The offset where the string was found.
+ * 
+ * @return true  - The string was found;
+ *         false - the string was not found
+ */
+static bool findBytesInBuffer(const char * pBuffer, uint64_t _bufferSize,
+                  const std::string& _searchString, 
+                  unsigned int& _foundOffset) {
+
+  // Initialize return values
+  _foundOffset = 0;
+
+  // Working variables
+  unsigned int stringLength = (unsigned int) _searchString.length();
+  unsigned int matchIndex = 0;
+
+  for (unsigned int index = 0; index < _bufferSize; ++index) {
+    char aChar = pBuffer[index];
+    if (aChar == _searchString[matchIndex++]) {
+      if (matchIndex == stringLength) {
+        _foundOffset = 1 + index - stringLength;
+        return true;
+      }
+    } else {
+      matchIndex = 0;
+    }
+  }
+
+  return false;
+}
+
+static void remove_xsabin_mirror(void * pXsaBinBuffer)
+{
+  static const std::string MIRROR_DATA_START = "XCLBIN_MIRROR_DATA_START";
+  static const std::string MIRROR_DATA_END = "XCLBIN_MIRROR_DATA_END";
```

Comment:
Please don't use UPPER_CASE for variables, reserved for macros (which we should rarely use).